docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+Provide that attribute in a `RegisterPostInitializationOutput` step. Register for callbacks on the classes with
+`CreateSyntaxProvider` using the predicate `node is ClassDeclarationSyntax classDeclarationSyntax && classDeclarationSyntax.BaseList is not null && classDeclarationSyntax.BaseList.Types.Count > 0` to collect the classes that possibly implements at least one interface, And use tuples (or create an equatable model) to pass along that information.
+
+**Example:**
+
+```csharp
+public interface IUserInterface
+{
+    public int InterfaceProperty { get; set; }
+}
+
+public interface IUserInterface2
+{
+    public float InterfacePropertyOnlyGetter { get; }
+}
+```
+
+```csharp
+[AutoImplement(nameof(IUserInterface), nameof(IUserInterface2))]
+public partial class UserClass
+{
+    public string UserProp { get; set; }
+}
+```
+
+```csharp
+#nullable enable
+[Generator]
+public class AutoImplementGenerator : IIncrementalGenerator
+{
+    private const string NameOfString = "nameof(";
+    private const string AutoImplementAttributeNameSpace = "AttributeGenerator";
+    private const string AutoImplementAttributeName = "AutoImplement";
+    private const string AutoImplementAttributeClassName = $"{AutoImplementAttributeName}Attribute";
+    private const string FullyQualifiedMetadataName = $"{AutoImplementAttributeNameSpace}.{AutoImplementAttributeClassName}";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(ctx =>
+        {
+            //Generate the AutoImplement Attribute
+            const string autoImplementAttributeDeclarationCode = $$"""
+using System;
+namespace {{AutoImplementAttributeNameSpace}};
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+sealed class {{AutoImplementAttributeClassName}} : Attribute
+{
+    public string[] InterfacesNames { get; }
+    public {{AutoImplementAttributeClassName}}(params string[] interfacesNames)
+    {
+        InterfacesNames = interfacesNames;
+    }
+}
+""";
+            ctx.AddSource($"{AutoImplementAttributeClassName}.g.cs", autoImplementAttributeDeclarationCode);
+        });
+
+        IncrementalValuesProvider<Model> provider = context.SyntaxProvider.ForAttributeWithMetadataName(
+            fullyQualifiedMetadataName: FullyQualifiedMetadataName,
+            predicate: static (node, cancellationToken_) => node is ClassDeclarationSyntax,
+            transform: static (ctx, cancellationToken) =>
+            {
+                ClassDeclarationSyntax classDeclarationSyntax = (ClassDeclarationSyntax)ctx.TargetNode;
+                
+                string className = classDeclarationSyntax.Identifier.ValueText;
+                string classNameSpace = GetClassNameSpace(classDeclarationSyntax.Parent);
```

Comment:
Don't do either of these things. The context has the type symbol for the attributed class; get this info directly from the symbol, not from syntax. This code will not work robustly; it won't handle nested types, types in the global namespace, or syntax that deviates from the standard single-namespace form.